x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.test.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mockConfig, mockLogger } from '../__mocks__';
+
+import { createEnterpriseSearchRequestHandler } from './enterprise_search_request_handler';
+
+jest.mock('node-fetch');
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetchMock = require('node-fetch') as jest.Mock;
+const { Response } = jest.requireActual('node-fetch');
+
+const responseMock = {
+  ok: jest.fn(),
+  notFound: jest.fn(),
+  customError: jest.fn(),
+};
+const KibanaAuthHeader = 'Basic 123';
+
+describe('createEnterpriseSearchRequestHandler', () => {
+  beforeEach(() => {
+    responseMock.ok.mockClear();
+    responseMock.notFound.mockClear();
+    responseMock.customError.mockClear();
+    fetchMock.mockReset();
+  });
+
+  it('makes an API call and returns the response', async () => {
+    const responseBody = {
+      results: [{ name: 'engine1' }],
+      meta: { page: { total_results: 1 } },
+    };
+
+    EnterpriseSearchAPI.mockReturn(responseBody);
+
+    const requestHandler = createEnterpriseSearchRequestHandler({
+      config: mockConfig,
+      log: mockLogger,
+      path: '/as/credentials/collection',
+    });
+
+    await makeAPICall(requestHandler, {
+      query: {
+        type: 'indexed',
+        pageIndex: 1,
+      },
+    });
+
+    EnterpriseSearchAPI.shouldHaveBeenCalledWith(
+      'http://localhost:3002/as/credentials/collection?type=indexed&pageIndex=1'
+    );
+
+    expect(responseMock.ok).toHaveBeenCalledWith({
+      body: responseBody,
+    });
+  });
+
+  describe('when an API request fails', () => {
+    it('should return 502 with a message', async () => {
+      EnterpriseSearchAPI.mockReturnError();
+
+      const requestHandler = createEnterpriseSearchRequestHandler({
+        config: mockConfig,
+        log: mockLogger,
+        path: '/as/credentials/collection',
+      });
+
+      await makeAPICall(requestHandler);
+
+      EnterpriseSearchAPI.shouldHaveBeenCalledWith(
+        `http://localhost:3002/as/credentials/collection`
```

Comment:
Sorry, I totally noticed this last night and forgot to leave a comment - this isn't a template literal so we can just use a regular string

```suggestion
        'http://localhost:3002/as/credentials/collection'
```